File: App/ClientApp/src/app/Features/cart/interfaces/sales-transaction.ts
import { ICartRequest } from "./cart-request";
import { CreditCard } from "./credit-card";
import { BillingInfo } from "./billing-info";

export interface SalesTransaction
{
  orderId: string;
  billingInfo: BillingInfo;
  card: CreditCard;
  requests: ICartRequest[]
}

File: App/Shared/DTOs/SalesTransaction.cs
using App.Shared.Interfaces;

namespace App.Shared.DTOs;

public class SalesTransaction
{
    public string? OrderId { get; set; }
    public BillingInfo? BillingInfo { get; set; }
    public CreditCard? Card { get; set; }
    public IList<CartRequest>? Requests { get; set; }
}